packages/shared-ux/avatar/solution/src/solution_avatar.tsx

```diff
@@ -0,0 +1,82 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+import './solution_avatar.scss';
+
+import React from 'react';
+import classNames from 'classnames';
+
+import { DistributiveOmit, EuiAvatar, EuiAvatarProps, IconType } from '@elastic/eui';
+import { EuiIconType } from '@elastic/eui/src/components/icon/icon';
+
+type FilterStartingWith<Set, Needle extends string> = Set extends `${Needle}${infer _X}`
+  ? Set
+  : never;
+
+type ExtractSolutionName<SolutionString extends string> =
+  SolutionString extends `logo${infer Solution}` ? Solution : { error: 'Cannot parse logo' };
+
+type SolutionNameType = ExtractSolutionName<FilterStartingWith<EuiIconType, 'logo'>>;
```

Comment:
This reminds me though that we do have a primary solution that will always have to override the `iconType` because it is two words. `Enterprise search`. Is there anyway to de-camelCase to the results of the logo types to compare against?

Reply:
Managed to fix this.